### PR TITLE
Document MQTT lightweight and API functions

### PR DESCRIPTION
**Pull request by gordonwang0**
Adds documentation for currently implemented functions.

### CODE REVIEW

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 1]
```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
 struct MQTTPacketInfo;
 typedef struct MQTTPacketInfo MQTTPacketInfo_t;
 
+/**
+ * @brief Signature of the transport interface receive function.
+ *
+ * A function with this signature must be provided to the MQTT library to read
+ * data off the network.
+ *
+ * @param[in] context The context provided with this function.
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:57]: The context --> The MQTT context.

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 2]
```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
...
-    MQTTQoS2 = 2
+    MQTTQoS0 = 0,    /**< Delivery at most once. */
+    MQTTQoS1 = 1,    /**< Delivery at least once. */
+    MQTTQoS2 = 2     /**< Delivery exactly once. */
 } MQTTQoS_t;
 
+/**
+ * @brief Buffer passed to MQTT library.
+ *
+ * These buffers are not copied and must remain in scope for the duration of the
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:95]: The ones provided in init (network buffers) must remain in scope all the time the context is used?

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 3]
```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
...
 
+/**
+ * @brief MQTT CONNECT packet parameters.
+ */
 struct MQTTConnectInfo
 {
+    /**
+     * @brief Whether to establish a new, clean session or resume a previous session.
+     */
     bool cleanSession;
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:112]: Given the C89 guidelines, we probably cannot use bool. Not required in this PR though.

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 4]
```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
...
+ */
 struct MQTTSubscribeInfo
 {
+    /**
+     * @brief Quality of Service for subscription.
+     */
     MQTTQoS_t qos;
+
+    /**
+     * @brief Topic filter for subscription.
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:161]: Topic filter to subscribe to.

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 5]
```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
...
+     */
     MQTTQoS_t qos;
+
+    /**
+     * @brief Whether this is a retained message.
+     */
     bool retain;
+
+    /**
+     * @brief Topic name for message.
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:187]: Topic name on which the message is published.

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 6]
```diff
@@ -23,27 +23,57 @@
...
+ */
 #define UINT8_SET_BIT( x, position )    ( ( x ) = ( uint8_t ) ( ( x ) | ( 0x01U << ( position ) ) ) )
 
+/**
+ * @brief Get the high byte of a 16-bit integer.
+ */
 #define UINT16_HIGH_BYTE( x )    ( ( uint8_t ) ( ( x ) >> 8 ) )
+
+/**
+ * @brief Get the low byte of a 16-bit signed integer.
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:71]: signed -> unsigned

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 7]
```diff
@@ -23,27 +23,57 @@
...
+ */
 #define MQTT_DISCONNECT_REMAINING_LENGTH    ( ( uint8_t ) 0 )
 
+/**
+ * @brief Set a bit in an 8-bit unsigned integer.
+ */
 #define UINT8_SET_BIT( x, position )    ( ( x ) = ( uint8_t ) ( ( x ) | ( 0x01U << ( position ) ) ) )
 
+/**
+ * @brief Get the high byte of a 16-bit integer.
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:66]: a 16-bit integer --> a 16-bit unsigned integer

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 1 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:57]: This is actually the network context, such as socket (int).  Updated wording.

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 2 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:95]: Yes

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 4 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:161]: changed

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 5 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:187]: changed

File: libraries/standard/mqtt/src/mqtt_lightweight.c (hunk 6 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/src/mqtt_lightweight.c:71]: fixed

File: libraries/standard/mqtt/src/mqtt_lightweight.c (hunk 7 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/src/mqtt_lightweight.c:66]: fixed

**Pull request merged**